packages/pwa/app/routes.jsx

```diff
@@ -81,7 +81,8 @@ const routes = [
     },
     {
         path: '/product/:productId',
-        component: ProductDetail
+        component: ProductDetail,
+        localized: true
```

Comment:
This is the API for url customization feature. A boolean flag in the route config to indicate whether a route path should include locale.

Another approach worth thinking about would be to accept path callback function and the SDK calls the function to determine the paths, but this means the route configs are no longer compatible with React router directly:
{
        path: (locale) => `${locale}/product/:productId`,
        component: ProductDetail
}

Reply:
@kevinxh How does this approach look like if we want to add another prefix path such siteId, siteIdAlias? 

 On the SDK point of view, the locale does make sense, but having `siteId` or `siteAlias` does not seem so.

Reply:
well, you add a new flag `multisite:true` just like `locale`, and pass the siteId to the SDK and let the SDK's switch component handle everything for ya.

Reply:
Sorry, what I meant to ask is what does multisite mean to the SDK? 🤔 .

I did previously have a convo with @bendvc about a similar approach, but he kinda disagreed on this idea as it makes the SDK to be very opinionated.

Reply:
it doesn't have to be a concept of `multisite` from sdk's perspective. Instead of a flag like `multisite`, you could maybe take a `basename` property.

The point that this PR is trying to make is, we can implement the url customization feature in our SDK, potentially in the `Switch` component, instead of us just trying to use `basename` from react router, because that basename is not what we want, it has the limitation of forcing every single route to have a basename.